src/components/AppNavigation/CalendarShare.vue

```diff
@@ -95,48 +102,149 @@ export default {
-							displayName: result.displayname,
-							icon: isGroup ? 'icon-group' : 'icon-user',
-							uri: result.principalScheme,
-							isGroup,
-						})
-					}
-					return list
-				}, [])
+				const davPromise = this.findShareesFromDav(query, hiddenPrincipalSchemes, hiddenUrls)
+				const ocsPromise = this.findShareesFromCircles(query, hiddenPrincipalSchemes, hiddenUrls)
+
+				const [davResults, ocsResults] = await Promise.all([davPromise, ocsPromise])
+				this.usersOrGroups = [
+					...davResults,
+					...ocsResults,
+				]
+
 				this.isLoading = false
 				this.inputGiven = true
 			} else {
 				this.inputGiven = false
 				this.isLoading = false
 			}
+		}, 500),
+		/**
+		 *
+		 * @param {String} query The search query
+		 * @param {String[]} hiddenPrincipals A list of principals to exclude from search results
+		 * @param {String[]} hiddenUrls A list of urls to exclude from search results
+		 * @returns {Promise<Object[]>}
+		 */
+		async findShareesFromDav(query, hiddenPrincipals, hiddenUrls) {
+			let results
+			try {
+				results = await client.principalPropertySearchByDisplayname(query)
+			} catch (error) {
+				return []
+			}
+
+			return results.reduce((list, result) => {
+				if (hiddenPrincipals.includes(decodeURI(result.principalScheme))) {
+					return list
+				}
+				if (hiddenUrls.includes(result.url)) {
+					return list
+				}
+
+				// Don't show resources and rooms
+				if (!['GROUP', 'INDIVIDUAL'].includes(result.calendarUserType)) {
+					return list
+				}
+
+				const isGroup = result.calendarUserType === 'GROUP'
+				list.push({
+					user: result[isGroup ? 'groupId' : 'userId'],
+					displayName: result.displayname,
+					icon: isGroup ? 'icon-group' : 'icon-user',
+					uri: result.principalScheme,
+					isGroup,
+					isCircle: false,
+					isNoUser: isGroup,
+					search: query,
+				})
+				return list
+			}, [])
+		},
+		/**
+		 *
+		 * @param {String} query The search query
+		 * @param {String[]} hiddenPrincipals A list of principals to exclude from search results
+		 * @param {String[]} hiddenUrls A list of urls to exclude from search results
+		 * @returns {Promise<Object[]>}
+		 */
+		async findShareesFromCircles(query, hiddenPrincipals, hiddenUrls) {
+			let results
+			try {
+				results = await HttpClient.get(generateOcsUrl('apps/files_sharing/api/v1') + 'sharees', {
+					params: {
+						format: 'json',
+						search: query,
+						perPage: 200,
+						itemType: 'principals',
+					},
+				})
+				console.debug(results)
+			} catch (error) {
+				return []
+			}
+
+			if (results.data.ocs.meta.status === 'failure') {
+				return []
+			}
+			let circles = []
+			if (Array.isArray(results.data.ocs.data.circles)) {
+				circles = circles.concat(results.data.ocs.data.circles)
+			}
+			if (Array.isArray(results.data.ocs.data.exact.circles)) {
+				circles = circles.concat(results.data.ocs.data.exact.circles)
+			}
+
+			if (circles.length === 0) {
+				return []
+			}
```

Comment:
We really need an api for this....

Reply:
Hm, it's a bit messy and also leads to errors, see https://github.com/nextcloud/calendar/issues/2220.
But that's how it is for now.